File: DTOF.php
<?php
// Establish database connection
$servername = "your_servername";
$username = "your_username";
$password = "your_password";
$dbname = "your_dbname";

try {
    $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password);
    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
} catch(PDOException $e) {
    echo "Connection failed: " . $e->getMessage();
}

// Fetch all player stats or based on the provided player name
if (isset($_GET['player_name'])) {
    $playerName = $_GET['player_name'];
    
    $stmt = $conn->prepare("SELECT * FROM PlayerStats WHERE league_name LIKE :playerName");
    $stmt->bindValue(':playerName', "%$playerName%", PDO::PARAM_STR);
    $stmt->execute();
    $playerStats = $stmt->fetchAll(PDO::FETCH_ASSOC);
} else {
    $stmt = $conn->prepare("SELECT * FROM PlayerStats");
    $stmt->execute();
    $playerStats = $stmt->fetchAll(PDO::FETCH_ASSOC);
}
?>

<!DOCTYPE html>
<html>
<head>
    <title>League of Legends Stats</title>
    <!-- Your CSS styling here -->
</head>
<body>
    <h1>League of Legends Stats</h1>

    <!-- Search bar -->
    <div class="search-bar">
        <form action="<?php echo $_SERVER['PHP_SELF']; ?>" method="GET">
            <input type="text" name="player_name" placeholder="Search by Player Name">
            <input type="submit" value="Search">
        </form>
    </div>

    <!-- Table of player stats -->
    <table border="1">
        <thead>
            <tr>
                <th>League Name</th>
                <th>Games Played</th>
                <th>Wins</th>
                <th>Losses</th>
                <th>Win Rate</th>
                <th>Kills</th>
                <th>Deaths</th>
                <th>Assists</th>
                <th>KDA</th>
                <th>CS</th>
                <th>CS/M</th>
                <th>Gold</th>
                <th>Gold/M</th>
                <th>Damage</th>
                <th>Damage/M</th>
                <th>KPAR</th>
                <th>KS</th>
                <th>GS</th>
                <th>CP</th>
                <th>Champions</th>
                <!-- Add headers for other columns -->
            </tr>
        </thead>
        <tbody>
            <?php foreach ($playerStats as $stat): ?>
                <tr>
                    <!-- Display player stats within table rows -->
                    <td><?php echo $stat['league_name']; ?></td>
                    <td><?php echo $stat['games_played']; ?></td>
                    <td><?php echo $stat['wins']; ?></td>
                    <td><?php echo $stat['losses']; ?></td>
                    <td><?php echo $stat['win_rate']; ?></td>
                    <td><?php echo $stat['kills']; ?></td>
                    <td><?php echo $stat['deaths']; ?></td>
                    <td><?php echo $stat['assists']; ?></td>
                    <td><?php echo $stat['kda']; ?></td>
                    <td><?php echo $stat['cs']; ?></td>
                    <td><?php echo $stat['cs_per_min']; ?></td>
                    <td><?php echo $stat['gold']; ?></td>
                    <td><?php echo $stat['gold_per_min']; ?></td>
                    <td><?php echo $stat['damage']; ?></td>
                    <td><?php echo $stat['damage_per_min']; ?></td>
                    <td><?php echo $stat['kpar']; ?></td>
                    <td><?php echo $stat['ks']; ?></td>
                    <td><?php echo $stat['gs']; ?></td>
                    <td><?php echo $stat['cp']; ?></td>
                    <td><?php echo $stat['champions']; ?></td>
                    <!-- Display other stats -->
                </tr>
            <?php endforeach; ?>
        </tbody>
    </table>
</body>
</html>

File: DTOF.py
from flask import Flask, render_template
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'your_database_uri_here'  # Replace with your actual database URI
db = SQLAlchemy(app)

# Define a model for your player stats table
class PlayerStats(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    league_name = db.Column(db.String(100))
    games_played = db.Column(db.Integer)
    wins = db.Column(db.Integer)
    losses = db.Column(db.Integer)
    win_rate = db.Column(db.Float)
    kills = db.Column(db.Float)
    deaths = db.Column(db.Float)
    assists = db.Column(db.Float)
    kda = db.Column(db.Float)
    cs = db.Column(db.Float)
    cs_per_min = db.Column(db.Float)
    gold = db.Column(db.Float)
    gold_per_min = db.Column(db.Float)
    damage = db.Column(db.Float)
    damage_per_min = db.Column(db.Float)
    kpar = db.Column(db.Float)
    ks = db.Column(db.Float)
    gs = db.Column(db.Float)
    cp = db.Column(db.Float)
    champions = db.Column(db.Integer)
    # Define other columns for each statistic field

# Route to fetch stats from the database
@app.route('/')
def landing_page():
    stats_from_db = PlayerStats.query.all()  # Retrieve stats from the database
    return render_template('landing_page.html', player_stats=stats_from_db)
@app.route('/search', methods=['GET'])
def search_player():
    player_name = request.args.get('player_name', '')
    
    if player_name:
        # Query the database for the player stats using the provided name
        stats_from_db = PlayerStats.query.filter_by(league_name=player_name).all()
    else:
        # If no name provided, display all player stats
        stats_from_db = PlayerStats.query.all()

    return render_template('landing.html', player_stats=stats_from_db)

if __name__ == '__main__':
    app.run(debug=True)

if __name__ == '__main__':
    app.run(debug=True)
